app/code/Magento/CatalogUrlRewrite/Observer/ProductProcessUrlRewriteSavingObserver.php

```diff
@@ -6,56 +6,108 @@
 namespace Magento\CatalogUrlRewrite\Observer;
 
 use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\ProductRepository;
+use Magento\CatalogUrlRewrite\Model\ProductScopeRewriteGenerator;
 use Magento\CatalogUrlRewrite\Model\ProductUrlPathGenerator;
 use Magento\CatalogUrlRewrite\Model\ProductUrlRewriteGenerator;
-use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\UrlRewrite\Model\Exception\UrlAlreadyExistsException;
+use Magento\UrlRewrite\Model\Storage\DeleteEntitiesFromStores;
 use Magento\UrlRewrite\Model\UrlPersistInterface;
 use Magento\Framework\Event\ObserverInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Store\Api\StoreWebsiteRelationInterface;
 
 /**
  * Class ProductProcessUrlRewriteSavingObserver
+ *
+ * Observer to update the Rewrite URLs for a product.
+ * This observer is triggered on the save function when making changes
+ * to the products website on the Product Edit page.
  */
 class ProductProcessUrlRewriteSavingObserver implements ObserverInterface
 {
     /**
-     * @var ProductUrlRewriteGenerator
+     * @var ProductUrlRewriteGenerator $productUrlRewriteGenerator
```

Comment:
Why did you added variable name here?It is needed only for inline blocks.

Reply:
I was never sure whether it was required or not. There were some entries where they were there, and some where they weren't. I was just making them consistent. I see now I went the wrong way on this one :). I'll remove them to make it ahem.... consistent with the rest of Magento :)